src/main/groovy/com/palantir/launchconfig/EclipseLaunchConfigTask.groovy

```diff
@@ -15,16 +15,22 @@ class EclipseLaunchConfigTask extends DefaultTask {
 
     void setup() {
         this.project.tasks.withType(JavaExec) { javaExec ->
-            Path launchFile = Paths.get("${project.projectDir}/${project.name}-${javaExec.name}.launch")
-            this.outputs.file(launchFile.toString())
+            if (isIncluded(javaExec.name)) {
+                Path launchFile = Paths.get("${project.projectDir}/${project.name}-${javaExec.name}.launch")
+                this.outputs.file(launchFile.toString())
+            }
         }
     }
 
     @TaskAction
     void generate() {
         this.project.tasks.withType(JavaExec) { javaExec ->
-            logger.debug("Generating launch file for '%s'.", javaExec.name)
-            writeLaunchFile(javaExec)
+            if (isIncluded(javaExec.name)) {
+	            logger.debug("Generating launch file for '%s'.", javaExec.name)
+	            writeLaunchFile(javaExec)
+			} else {
+                logger.debug("Skipping generation of '%s' since it is excluded in the launchConfig.", javaExec.name)
```

Comment:
there's also a bug here with the `%s`. check out issue #2 if you can fix that :-D